### ISSUE TITLE
Rayman Arena - broken graphics in the menu and an exception after that

### ISSUE THREAD

**Issue by furrya** [Cxbx-Reloaded/Cxbx-Reloaded#236]
Rayman Arena has broken graphics menu after loading screen 
![screenshot_1](https://cloud.githubusercontent.com/assets/25540784/23930337/15fe7448-096f-11e7-9dc6-4195cdb4c629.png)
![screenshot_5](https://cloud.githubusercontent.com/assets/25540784/23930459/b634b044-096f-11e7-85e3-0aa08c2c59e3.png)
![screenshot_4](https://cloud.githubusercontent.com/assets/25540784/23930458/b632f9de-096f-11e7-8f7e-4a244ac94de2.png)
And sometimes
![screenshot_6](https://cloud.githubusercontent.com/assets/25540784/23930460/b63547ca-096f-11e7-9a6d-14fd41bba40c.png)

And XBE Dump Below
[Xbe.txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/843224/Xbe.txt)

**Comment by furrya** [comment]
EmuCR-Cxbx-Reloaded-20170316 .. Hmm now the exception code appeared and nothing else 
![screenshot_8](https://cloud.githubusercontent.com/assets/25540784/24007551/46d37c50-0ab2-11e7-9959-326248290911.png)
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/848399/KrnlDebug.txt)

**Comment by LukeUsher** [comment]
Can you upload a KrnlDebug.txt for the version it was showing graphics on?

**Comment by PatrickvL** [comment]
Showing the build number is paying off!

**Comment by furrya** [comment]
@LukeUsher  Hmm. Very strange behaviour of the appearing exception code i see right now,  Sometimes it is appeared from start launching the game and occasionally the game  shows the intro and Rayman can reach the menu like that 

And KrnlDebug.txt with graphics below 
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/848530/KrnlDebug.txt)
![screenshot_10](https://cloud.githubusercontent.com/assets/25540784/24009671/0472cf58-0ab9-11e7-83a5-50ebb558d7db.png)

**Comment by furrya** [comment]
Now it"s regressed at the loading screen 
![screenshot_1](https://cloud.githubusercontent.com/assets/25540784/24337190/3c5805fe-12d5-11e7-92fb-fa5932b51369.png)
![screenshot_1](https://cloud.githubusercontent.com/assets/25540784/24337192/426080de-12d5-11e7-968c-c19ef8dc9a11.png)

**Comment by LukeUsher** [comment]
I'm sorry about that, any change in the XDK signatures that allows new titles to work has potential to cause other titles to regress... We hope to have an automated testing platform to catch things like this fairly soon, we should see more stable results after that.

**Comment by jarupxx** [comment]
This seems to crash with 54be6c85. Creation VertexBuffer has failed. One old commit 201313b4 works well.
54be6c85 generates following logs.
```
[0x1F58] EmuD3D8: EmuIDirect3DResource8_Register
(
   pThis               : 0x046BA5D0 (->Data : 0x00000000)
   pBase               : 0x800390CC
);
[0x1F58] EmuIDirect3DResource8_Register : Creating VertexBuffer...
[0x1F58] MemoryManager.cpp: QueryAllocationSize(
   block                : 800390CC
);
[0x1F58] MemoryManager.cpp: IsAllocated(
   block                : 800390CC
);
[0x1F58] MemoryManager.cpp: IsAllocated returns 0
[0x1F58] EmuWarn: MemoryManager: Attempted to query memory that was not allocated via MemoryManager
[0x1F58] MemoryManager.cpp: QueryAllocationSize returns 0
[0x1F58] EmuWarn: CreateVertexBuffer Failed!

VB Size = 0x0

Error: 
Desc: 
[0x1F58] EmuD3D8: EmuIDirect3DVertexBuffer8_Lock2
(
   ppVertexBuffer      : 0x046BA5C0
   Flags               : 0x00000020
);
[0x1F58] EmuWarn: ppVertexBuffer->EmuVertexBuffer8 == NULL!
[0x1F58] EmuWarn: EmuX86_Write32(0x00000008, 0x00000000) [Unknown address]
[0x1F58] EmuWarn: EmuX86_Write32(0x0000000C, 0x3F7FFF58) [Unknown address]
```
[54be6c85-Trace.txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/881527/54be6c85-Trace.txt)
[201313b4-Trace.txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/881526/201313b4-Trace.txt)

**Comment by PatrickvL** [comment]
"Attempted to query memory that was not allocated via MemoryManager" is a side-effect of the new memory-manager that's currently lacking a feature (the ability to query within a pre-allocated memory block)

**Comment by PatrickvL** [comment]
We're aware of this issue though - see https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/issues/283
I'm trying to improve upon it, but I don't expect a quick fix is possible at the moment.

**Comment by PatrickvL** [comment]
@jarupxx Has PR #283 restored previous behaviour for this title?

**Comment by jarupxx** [comment]
@PatrickvL Unfortunately no. I tested cdc5d5cd still crashed.

**Comment by PatrickvL** [comment]
Bummer. Then the regression is probably caused by an OOVPA lookup mismatch - we should compare old with new and see which one(s) is/are different.

**Comment by LukeUsher** [comment]
Ooh that's interesting, I'll fix this today

**Comment by PatrickvL** [comment]
Ah, found the first real difference :

Old :
```
[0x18E8] EmuIDirect3DResource8_Register : Creating VertexBuffer...
[0x18E8] EmuIDirect3DResource8_Register : Successfully Created VertexBuffer (0x06E109A0)
```
New :
```
[0x1F58] EmuIDirect3DResource8_Register : Creating VertexBuffer...
[0x1F58] MemoryManager.cpp: QueryAllocationSize(
   block                : 8001BC04
);
[0x1F58] MemoryManager.cpp: IsAllocated(
   block                : 8001BC04
);
[0x1F58] MemoryManager.cpp: IsAllocated returns 0
[0x1F58] EmuWarn: MemoryManager: Attempted to query memory that was not allocated via MemoryManager
[0x1F58] MemoryManager.cpp: QueryAllocationSize returns 0
[0x1F58] EmuWarn: CreateVertexBuffer Failed!
```

**Comment by PatrickvL** [comment]
@jarupxx Can this be re-tested once PR #301 is merged in?

**Comment by jarupxx** [comment]
Improvement to creation VertexBuffer succeeded in cf4ff0cc, but crashed.
![cf4ff0cc](https://cloud.githubusercontent.com/assets/20138367/24570861/d0ff7526-16a8-11e7-89de-8d5e6f0ff89f.jpg)

**Comment by jarupxx** [comment]
Fixed in e250b1f8. Reached the menu! It crashes before going to the game.
![rayman arena](https://cloud.githubusercontent.com/assets/20138367/25559909/c9b0fe22-2d80-11e7-9aaa-4d8e39dda172.jpg)
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/966473/KrnlDebug.txt)

**Comment by furrya** [comment]
Cxbx-Reloaded Version 9173d4ab (May  3 2017) ' the game crashes at loading screen now
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/975327/KrnlDebug.txt)

![screenshot_5](https://cloud.githubusercontent.com/assets/25540784/25692691/7a958b5c-30df-11e7-9cc0-afce16e204b8.png)

**Comment by furrya** [comment]
Rayman Arena warning, after that warning the game crashes immediately   Cxbx Reloaded 488db560.
![screenshot_1](https://user-images.githubusercontent.com/25540784/28965359-af26c408-794b-11e7-85a7-c4efd516f0f2.png)
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/1200024/KrnlDebug.txt)

**Comment by RadWolfie** [comment]
I... am unable to produce it since I am stuck with the same previous error crash you're having. "0x000F99F2(=timeKillEvent+0x8c64)". If you are able to reproduce it every time you boot up the game, then I will work on it if I can get past the crash I mention recently.

**Comment by furrya** [comment]
@RadWolfie Oh, Ok. Thank you. I looked more closely when happens that error while loading. Hm, and yes -  sometimes another error has to appear
![screenshot_1](https://user-images.githubusercontent.com/25540784/28976963-01661c62-797b-11e7-85db-43909328f30d.png)
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/1200971/KrnlDebug.txt)

**Comment by RadWolfie** [comment]
No worry, I have discover 3 other titles has regress for "DirectSoundBuffer Lock Failed!" error. I'll try to find a fix and create new pull request within 12 hours from this post.

**Comment by PatrickvL** [comment]
This issue was moved to Cxbx-Reloaded/game-compatibility#183
